compiler/modules/nback/scaffold.bal

```diff
@@ -37,20 +37,37 @@ const BASE_REPR_TAGGED = "BASE_REPR_TAGGED";
 type BaseRepr UniformBaseRepr|BASE_REPR_TAGGED;
 type RetBaseRepr BaseRepr|BASE_REPR_VOID;
 
-type UniformRepr readonly & record {|
-    UniformBaseRepr base;
-    llvm:SingleValueType llvm;
-    t:UniformTypeBitSet subtype?;
+// Maps int to i1
+type BooleanRepr readonly & record {|
```

Comment:
I think it might be helpful to have a ReprFields record with base/llvm/isImmediate fields and thenn do `*ReprFields;` in each *Repr to make subtyping clearer.